src/app/posts/posts.component.ts

```diff
@@ -54,4 +60,38 @@ export class PostsComponent {
 
     this.router.navigate(['login']);
   }
+
+  submitLogout()
+  {
+    this.access_token = localStorage.getItem('access_token') ?? '';
+    this.user_name = localStorage.getItem('user_name') ?? '';
+
+    if (this.access_token !== '' && this.user_name !== '') {
+      this.user_name = this.user_name.split(" ")[0]
+
+      const url = 'http://localhost/api/auth/logout';
+      const headers = {
+        'Authorization': 'Bearer ' + this.access_token
+      }
+      const options = {                                                                                                                                                                                 
+        headers: new HttpHeaders(headers), 
+      };
+
+      this.http
+        .post<any>(url, '', options)
+        .pipe(
+          catchError( err => {return this.appModule.defineErrorsFromResponse(err, err.message, this)})
```

Comment:
if already passing the error object, dont think the err.message is needed, 'cause y can just access through the first